[esp32-C6 and esp-idf V5 branch] Fix ESP32-C6 GPIO pin validation and runtime compatibility

karan-vk opened the pull request:
## Summary
- Fix ESP32-C6 GPIO pin validation to allow GPIO 0-23 for LED data pins (GPIO 8 and others were incorrectly blocked)
- Fix multiple runtime crashes and compatibility issues for ESP32-C6 platform

## Problem
The V5-C6 branch had several issues preventing proper operation on ESP32-C6:

1. **Incorrect GPIO restrictions**: The `isPinOk()` function in `pin_manager.cpp` was missing ESP32-C6 specific handling, causing it to fall through to classic ESP32 logic which blocks GPIO 6-11 as "SPI flash pins". On ESP32-C6, the flash pins are actually GPIO 24-30, so GPIO 8 (and other low-numbered GPIOs) should be available for LED configuration.

2. **Runtime crashes**: IPAddress comparison with `0U` causing issues, array bounds violations with `selectedWiFi` index.

3. **Missing platform exclusions**: ESP32-C6 does not support capacitive touch sensors, RMT for NeoPixel (uses BitBang), or I2S1 parallel buses.

4. **ESPAsyncWebServer API differences**: The patched fork for C6 does not have `deferResponse()`.

## Changes

### Pin Manager (`pin_manager.cpp`)
- Added `#elif defined(CONFIG_IDF_TARGET_ESP32C6)` block to `isPinOk()`
- Correctly blocks GPIO 24-30 (SPI flash) instead of GPIO 6-11
- Optionally blocks GPIO 12-13 when USB-JTAG is enabled
- Notes strapping pins (8, 9, 15) in comments

### Bus Manager & Wrapper
- Added C6 exclusions for I2S1/parallel bus code paths
- Added BitBang method support for C6 (RMT not supported in NeoPixelBus for C6)
- Added early return in `esp32RMTInvertIdle()` for C6

### Runtime Fixes
- Fixed IPAddress comparison in `wled.cpp` to avoid null pointer issues
- Added bounds checking for `selectedWiFi` index
- Added `deferResponse()` wrapper function for C6 compatibility
- Updated AsyncWebServer constructor for C6 simpler API

### Build Configuration (`platformio.ini`)
- Updated library dependencies for IDF5 compatibility
- Disabled ESP-NOW (requires compatible AsyncTCP)
- Use patched FastLED, AsyncTCP, and ESPAsyncWebServer forks

## Features NOT Working on ESP32-C6

The following features are currently disabled or non-functional on ESP32-C6:

| Feature | Status | Reason |
|---------|--------|--------|
| **Infrared Remote** | Disabled | IRremoteESP8266 library not compatible with ESP32-C6 |
| **MQTT** | Disabled | AsyncMqttClient library not compatible with IDF5 (TODO: needs updated library) |
| **ESP-NOW Sync** | Disabled | QuickESPNow requires compatible AsyncTCP for IDF5 |
| **Alexa/Hue Emulation** | Disabled | Compile errors with current libraries |
| **WebSockets** | Disabled | Requires hacks in ESPAsyncWebServer for C6 |
| **Touch Button** | N/A | ESP32-C6 has no capacitive touch hardware (hardware limitation) |
| **Parallel I2S Output** | N/A | ESP32-C6 does not support I2S1 for LED output (hardware limitation) |
| **RMT LED Driver** | N/A | NeoPixelBus does not support RMT on C6; uses BitBang method instead |

### Hardware Limitations (ESP32-C6 vs other ESP32 variants)
- **Max 2 digital LED outputs** (same as ESP32-C3) vs 10+ on ESP32-S3
- **No capacitive touch pins** - touch buttons cannot be used
- **BitBang timing** - LED output uses software BitBang which may be less precise than RMT/I2S

### Software/Library Limitations (may be resolved in future)
- IR, MQTT, Alexa, WebSockets, and ESP-NOW require library updates for IDF5/C6 compatibility
- These are noted as TODOs in platformio.ini

## Testing
Successfully tested and verified working on:
- **ESP32-C6-DevKitC** (4MB)
- **Seeed Studio XIAO ESP32-C6**

Both boards now:
- Compile without errors
- Boot without crashes
- Allow GPIO 8 (and other valid GPIOs) to be configured for LED output
- Drive WS2812B LEDs correctly using BitBang method
- WiFi connectivity works
- Web UI is fully functional
- JSON API works
- Presets and effects work correctly

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added support for ESP32-C6 microcontroller with optimized hardware configuration.

* **Bug Fixes**
  * Improved WiFi network selection stability with enhanced boundary validation.
  * Refined pin management and device compatibility checks.

* **Chores**
  * Updated library dependencies for improved framework compatibility and device support.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

softhack007 commented on .vscode/extensions.json:
```diff
@@ -1,7 +1,6 @@
 {
-    // See http://go.microsoft.com/fwlink/?LinkId=827846
-    // for the documentation about the extensions.json format
```
this change is unrelated, please revert.

softhack007 commented on wled00/bus_wrapper.h:
```diff
@@ -245,7 +245,10 @@
 
 // RMT driver selection
-#if !defined(WLED_USE_SHARED_RMT)  && !defined(__riscv)
+#if defined(CONFIG_IDF_TARGET_ESP32C6)
```
@willmmiles this is a kind of preprocessor hack, that actually hides " NeoEsp32BitBang" under the  NeoEsp32RmtMethod label. Not sure if we should do it this way, what do you think?

softhack007 commented on wled00/fcn_declare.h:
```diff
@@ -548,6 +548,17 @@ void serveJsonError(AsyncWebServerRequest* request, uint16_t code, uint16_t erro
 void serveSettings(AsyncWebServerRequest* request, bool post = false);
 void serveSettingsJS(AsyncWebServerRequest* request);
 
+// ESP32-C6 compatibility: deferResponse not available in patched ESPAsyncWebServer fork
```
@willmmiles what do you think?

willmmiles replied:
I think I'd like to see a PR to our web server with any necessary fixes, not a removal of the new features.

softhack007 replied:
that's a good point - could be that the V5 branch is still making reference to my own fork, where i've fixed a few incompatibilities more than a year ago. If V5 is still using my fork, no problem i can make a PR to our webserver.

softhack007 commented on wled00/json.cpp:
```diff
@@ -1124,7 +1124,7 @@ void serveJson(AsyncWebServerRequest* request)
   }
 
   if (!requestJSONBufferLock(17)) {
-    request->deferResponse();    
```
is this specific to C6? If unrelated, please revert.

softhack007 commented on wled00/wled.h:
```diff
@@ -903,7 +903,12 @@ WLED_GLOBAL bool ledStatusState _INIT(false); // the current LED state
 
 // server library objects
+#if defined(CONFIG_IDF_TARGET_ESP32C6)
+// ESP32-C6 uses patched ESPAsyncWebServer fork with simpler constructor
```
@willmmiles what do you think - is this specific to C6 or maybe specific for ESP-IDF v5?

softhack007 commented on wled00/wled_server.cpp:
```diff
@@ -312,7 +312,7 @@ void initServer()
     bool isConfig = false;
 
     if (!requestJSONBufferLock(14)) {
-      request->deferResponse();
```
same question - why?

karan-vk commented:
I have addressed the review comments in the latest commits:

- Reverted unrelated changes in \`.vscode/extensions.json\`
- Cleaned up \`platformio.ini\` (removed \`QuickDebug\`, reverted \`GifDecoder\` hash)
- Improved safety of \`selectedWiFi\` index calculation in \`network.cpp\`

The remaining changes (NeoPixelBus/ESPAsyncWebServer forks) are necessary for ESP32-C6 support on IDF5.

softhack007 commented on platformio.ini:
```diff
@@ -300,14 +301,22 @@ build_flags = -g
+  bitbank2/AnimatedGIF@^1.4.7
+  https://github.com/Aircoookie/GifDecoder#bc3af189b6b1e06946569f6b4287f0b79a860f8e
   https://github.com/netmindz/esp_dmx/#esp-idf-v5-fixes
-  ${env.lib_deps}
```
Let me think about it - basicially we should follow the same approach as the "V4" builds - with an env specific to ESP-IDF V5, and a C6 specific one that "extends=" the V5 environment.

softhack007 commented:
Btw, I think we can move from platformio to [pioarduino](https://github.com/pioarduino/platform-espressif32). C6 and P4 are not officially supported in platformio. But pioarduino does support the new boards.
